app/src/components/pages/FileHomePage.jsx

```diff
@@ -0,0 +1,87 @@
+"use client"
+
+import { useEffect, useState } from 'react'
+import { getFirestore, collection, getDocs } from 'firebase/firestore'
+import { getAuth } from "firebase/auth"
+import app from '../../firebase/firebase' // Import dell'istanza di Firebase
+import Header from '@/components/ui/Header'
+import {
+  Card,
+  CardContent,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card"
+import { Link } from 'react-router-dom'
+
+export default function DocumentLibrary() {
+  const [documents, setDocuments] = useState([])
+
+  useEffect(() => {
+    const db = getFirestore(app) // Inizializza Firestore
+    const fetchDocuments = async () => {
+      try {
+        const querySnapshot = await getDocs(collection(db, 'documents'))
+        const docs = querySnapshot.docs.map(doc => ({
+          ...doc.data(),  // Estrarre i dati dal documento
+          id: doc.id      // Aggiungere l'ID del documento
+        }))
+        setDocuments(docs) // Impostare i dati nello stato
+      } catch (error) {
+        console.error('Errore nel recupero dei documenti: ', error)
+      }
+    }
+    fetchDocuments()
+  }, [])
+
+  return (
+    <div className="min-h-screen bg-gradient-to-b from-[#178563] to-white">
+      {/* Header */}
+      <Header />
+
+      {/* Main Content */}
+      <main className="container mx-auto px-4 py-8">
+        <h2 className="text-2xl font-semibold text-white mb-6">Contenuti Media</h2>
+        
+        {/* Add Document Button */}
+        <div className="mb-6">
+          <Link
+            to="/addfile"
+            className="bg-[#178563] hover:bg-[#178563]/90 text-white px-4 py-2 rounded-md"
+          >
+            Aggiungi Documento
+          </Link>
+        </div>
```

Comment:
Mostrare una pagina che dice all'utente mentee, nel momento in cui cerca di caricare un contenuto, che non ha i permessi per farlo va bene come soluzione, ma non è quella ottimale, nel momento in cui un utente non ha i permessi per non fare quell'azione, sarebbe meglio non mostrargliela proprio, quindi di conseguenza è meglio non mostrare quel bottone nel momento in cui si è effettuato l'accesso come mentee